libcloudforensics/providers/azure/internal/account.py

```diff
@@ -217,3 +258,187 @@ def CreateDiskFromSnapshot(
+      sleep(5)  # Wait for the vhd to be imported in the Azure storage container
+
+    if not disk_name:
+      disk_name = common.GenerateDiskName(snapshot,
+                                          disk_name_prefix=disk_name_prefix)
+
+    # Create a new disk from the imported snapshot
+    creation_data = {
+        'location': region,
+        'creation_data': {
+            'source_uri': copied_blob.url,
+            'storage_account_id': storage_account_id,
+            'create_option': DiskCreateOption.import_enum
+        }
+    }
+
+    try:
+      request = self.compute_client.disks.create_or_update(
+          self.default_resource_group_name,
+          disk_name,
+          creation_data,
+          sku=disk_type)
+      while not request.done():
+        sleep(5)  # Wait 5 seconds before checking disk status again
+      disk = request.result()
+    except CloudError as exception:
+      raise RuntimeError('Could not create disk from URI {0:s}: {1:s}'
+                         .format(snapshot_uri, str(exception)))
+
+    # Cleanup the temporary account storage
+    self._DeleteStorageAccount(storage_account_name)
+
+    return compute.AZDisk(self,
+                          disk.id,
+                          disk.name,
+                          disk.location,
+                          disk.zones)
+
+  def _CreateStorageAccount(self,
+                            storage_account_name: str,
+                            region: Optional[str] = None) -> Tuple[str, str]:
+    """Create a storage account and returns its ID and access key.
+
+    Args:
+      storage_account_name (str): The name for the storage account.
+      region (str): Optional. The region in which to create the storage
+          account. If not provided, it will be created in the default_region
+          associated to the AZAccount object.
+
+    Returns:
+      Tuple[str, str]: The storage account ID and its access key.
+
+    Raises:
+      ValueError: If the storage account name is invalid.
+    """
+
+    if not common.REGEX_ACCOUNT_STORAGE_NAME.match(storage_account_name):
+      raise ValueError(
+          'Storage account name {0:s} does not comply with {1:s}'.format(
+              storage_account_name, common.REGEX_ACCOUNT_STORAGE_NAME.pattern))
+
+    if not region:
+      region = self.default_region
+
+    creation_data = {
+        'location': region,
+        'sku': {
+            'name': 'Standard_RAGRS'
+        },
+        'kind': 'Storage'
+    }
+
+    request = self.storage_client.storage_accounts.create(
+        self.default_resource_group_name,
+        storage_account_name,
+        creation_data
+    )
+    storage_account = request.result()
+    storage_account_keys = self.storage_client.storage_accounts.list_keys(
+        self.default_resource_group_name, storage_account_name)
+    storage_account_keys = {v.key_name: v.value
+                            for v in storage_account_keys.keys}
```

Comment:
Shouldn't `v` be `k`? You're accessing keys it seems, not values. Also, using `key` as a variable name would be better.

```suggestion
                            for key in storage_account_keys.keys}
```